src/Compilers/Core/Portable/RealParser.cs

```diff
@@ -0,0 +1,786 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// A set of utilities for converting from a decimal floating-point literal string to its IEEE float
+    /// or double representation, which coniders all digits signficant and correctly rounds according to
+    /// the IEEE round-to-nearest-ties-to-even mode. This code does not support a leading sign character,
+    /// as that is not part of the C# or VB floating-point literal lexical syntax.
+    /// 
+    /// If you change this code, please run the set of long-running random tests in the solution
+    /// RandomRealParserTests.sln. That solution is not included in Roslyn.sln as it is Windows-specific.
```

Comment:
It is a separate .sln (Solution) that is written using Windows-specific facilities.